ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -875,10 +876,17 @@ public void alterTable(String catName, String dbName, String tblName, Table newT
       if (newTbl.getParameters() != null) {
         newTbl.getParameters().remove(hive_metastoreConstants.DDL_TIME);
       }
-      newTbl.checkValidity(conf);
       if (environmentContext == null) {
         environmentContext = new EnvironmentContext();
       }
+      if (environmentContext.isSetProperties() && AlterTableType.RENAME.toString()
```

Comment:
could we do something like AlterTableType.valueOf(..) = AlterTableType.RENAME?

Reply:
**environmentContext.isSetProperties()** is a **NULL** check for properties map inside, if that isn't set the map inside is null, trying to fetch value from that would fetch us an **NPE**. 
done at all places:
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java#L462

have refactored to use`AlterTableType.valueOf(..)`